### PR TITLE
end2end - #226 rems integration and others

### PR DESCRIPTION
**Pull request by jLebioda**

### CODE REVIEW

**Review by vildead — APPROVED** [review]
tested using POST request.

Code looks nice from what I can tell.

File: core/models/user.py [hunk 1]
```diff
@@ -186,3 +187,30 @@ def can_edit_contract(self, contract):
...
+        Otherwise - it will raise an exception
+        """
+        notes = f'Set automatically by REMS application #{application}'
+        dataset = Dataset.objects.get(elu_accession=resource)
+        new_access = Access(
+            user=self,
+            dataset=dataset,
+            access_notes=notes
+        )
+        new_access.save()
```
Comment (pinarpink) [inline core/models/user.py:215]: Before we return can we add here:

```
 new_logbook_entry = Share(  dataset=dataset,            granted_on=datetime.now(), share_notes='rems-application:'+application)
new_logbook_entry.save
```

**Review by pinarpink — APPROVED** [review]
Thanks a lot @jLebioda. I had one comment on creating a logbook entry. Otherwise looks good to me.

File: core/models/user.py (hunk 1 above)
Comment (jLebioda) [reply core/models/user.py:215]: looks good!

File: core/migrations/0019_auto_20210616_1407.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    operations = [
+        migrations.AddField(
+            model_name='access',
+            name='user',
+            field=models.ForeignKey(blank=True, null=True, on_delete=django.db.models.deletion.CASCADE, related_name='user', to=settings.AUTH_USER_MODEL, verbose_name='User that has the access'),
+        ),
+        migrations.AlterField(
+            model_name='access',
+            name='defined_on_locations',
+            field=models.ManyToManyField(blank=True, help_text='The dataset locations on which access is defined.', null=True, related_name='accesses', to='core.DataLocation', verbose_name='Data Locations'),
```
[the remark this answers is not in the document]
Comment (jLebioda) [reply core/migrations/0019_auto_20210616_1407.py]: Exactly! Just corrected that in cff2f55
Thanks for spotting :)

**Comment by jLebioda** [comment]
@pinarpink @vildead just implemented both suggestions, let's wait for CI to confirm the tests

**Comment by jLebioda** [comment]
Ready to be reviewed again; I've included some other minor improvements (also from the previous MR #240):

 * added a modal when publishing a dataset (this will also publish relevant project/data_declaration etc.)
 * added unpublish buttons to Dataset, Partner, Project, Cohort (they will display a message "Do you want to continue? YES/NO" + "this will not unpublish the entry from other systems")
 * added a button "See in django-admin" for administrators on Dataset, Partner, Project, Cohort pages
 * the elu_accession is a read-only field for Dataset, Partner, Project, Cohort
 * the is_published field is completely hidden (there's "Publish"/"Unpublish" buttons and "See in django-admin")
 * when using docker-compose, the version on about and index pages will display "develop", and not "undefined" anymore
 * added editing PartnerRoles (solves #235)
 * added `linebreaks` filter to Project's lay summary and Dataset description (i.e. line breaks will be translated to <br>`)
 * added `API_KEY` field to User model (so that there is no need for looking for `GLOBAL_API_KEY` anymore, maybe in the future we could even remove it completely)
 * added `oidc_id` to User model
 * added both API_KEY and oidc_id to User form in django admin
 * added search box to datasets and projects in django admin
 * dataset and project pages in django admin will display more columns (so that it's not just 1 column with `__str__()`, but title, acronym, elu_id etc.

**Comment by vildead** [comment]
Great job @jLebioda! :1st_place_medal: 
I have added one humble commit :D 
I have tested the POST request and other features - works like a charm.

**Pull request merged**